lib/util/storage/database/database_service.dart

```diff
@@ -32,215 +38,211 @@ class DBException {
-      } else {
-        // Xprv.fromEncryptedXprv('xprv', 'password');
-        _dbService.unlocked = true;
+      switch (item.runtimeType) {
+        case Wallet:
+          await walletRepository.insertWallet(item, _database);
+          break;
+        case ValueTransferInfo:
+          await vttRepository.insertTransaction(item, _database);
+          break;
+        case Account:
+          await accountRepository.insertAccount(item, _database);
+          break;
+        default:
+          return false;
       }
-      return dbError;
-    } on DBException {
-      rethrow;
+    } catch (e) {
+      return false;
     }
+    return true;
   }
 
-  void dispose() {
-    _database.close();
-    _dbService._database.close();
-    _dbService._dbConfig = null;
-    _dbService.dispose();
+  Future<bool> delete(dynamic item) async {
+    try {
+      switch (item.runtimeType) {
+        case Wallet:
+          await walletRepository.deleteWallet(item.id, _database);
+          break;
+        case ValueTransferInfo:
+          await vttRepository.deleteTransaction(item.txnHash, _database);
+          break;
+        case Account:
+          await accountRepository.deleteAccount(item.address, _database);
+          break;
+        default:
+          return false;
+      }
+    } catch (e) {
+      return false;
+    }
+    return true;
   }
 
-  void lockDatabase() {
-    _database.close();
-    _dbService._dbConfig = null;
+  Future<bool> update(dynamic item) async {
+    try {
+      switch (item.runtimeType) {
+        case Wallet:
+          await walletRepository.updateWallet(item.id, _database);
+          break;
+        case ValueTransferInfo:
+          await vttRepository.updateTransaction(item.transactionId, _database);
+          break;
+        case Account:
+          await accountRepository.updateAccount(item, _database);
+          break;
+        default:
+          return false;
+      }
+    } catch (e) {
+      return false;
+    }
+    return true;
   }
 
-  Future<void> configure(
-      {required String path, required String password}) async {
-    if (_dbConfig == null) {
-      _dbConfig = _DBConfiguration(path: path, password: password);
-    } else {
-      _dbConfig = null;
-      _dbConfig = _DBConfiguration(path: path, password: password);
-    }
+  Future<bool> masterKeySet() async {
+    bool keyExists = await keyChain.keyExists(_database);
+    return keyExists;
   }
 
-  Future<String> openDatabase() async {
+  Future<bool> verifyPassword(String password) async {
     try {
-      return await _initDatabase();
-    } on DBException {
-      rethrow;
+      bool keyExists = await masterKeySet();
+      if (keyExists) {
+        String? key = await keyChain.getKey(_database);
+
+        bool valid = await keyChain.validatePassword(key, password);
+        if (valid) {
+          unlocked = true;
+        }
+        return valid;
+      }
+    } catch (e) {
+      return false;
     }
+    return false;
```

Comment:
IMHO this is easier to read. What do you think?
```suggestion
        if (!keyExists) {
          return false
        }
        
        String? key = await keyChain.getKey(_database);
  
        bool valid = await keyChain.validatePassword(key, password);
        if (valid) {
          unlocked = true;
        }
        return valid;
      }
    } catch (e) {
      return false;
    }

```